wicketd/src/update_tracker.rs

```diff
@@ -2434,13 +2799,130 @@ impl<'a> SpComponentUpdateContext<'a> {
+                const WAIT_FOR_BOOT_TIMEOUT: Duration = Duration::from_secs(30);
+
+                // We need to reset the RoT in order to check the signature on what we just
+                // updated
+                registrar
+                    .new_step(
+                        SpComponentUpdateStepId::Resetting,
+                        "Resetting the RoT to check the bootloader signature",
+                        move |_cx| async move {
+                            update_cx
+                                .reset_sp_component(SpComponent::ROT.const_as_str())
+                                .await
+                                .map_err(|error| {
+                                    SpComponentUpdateTerminalError::RotResetFailed {
+                                        error,
+                                    }
+                                })?;
+                            StepSuccess::new(()).into()
+                        },
+                    )
+                    .register();
+
+                registrar
+                    .new_step(
+                        SpComponentUpdateStepId::Resetting,
+                        "Waiting for RoT to boot".to_string(),
+                        move |_cx| async move {
+                            let (_, stage0next_error) = update_cx
+                                .wait_for_rot_boot_info(WAIT_FOR_BOOT_TIMEOUT)
+                                .await
+                                .map_err(|error| {
+                                    SpComponentUpdateTerminalError::GetRotBootInfoFailed { error }
+                                })?;
+
+                            // check that stage0next is valid before we try to set the component
+                            if let Some(error) = stage0next_error {
+                                return Err(SpComponentUpdateTerminalError::RotBootloaderError {
+                                    error: anyhow!(format!("{error:?}"))
+                                });
+                            }
+                            StepSuccess::new(()).into()
+                        },
+                    )
+                    .register();
+
+                // Actually set stage0 to use the new firmware
+                registrar
+                    .new_step(
+                        SpComponentUpdateStepId::SettingActiveBootSlot,
+                        format!("Setting {component_name} active slot to {firmware_slot}"),
+                        move |_cx| async move {
+                            update_cx
+                                .set_component_active_slot(
+                                    component_name,
+                                    firmware_slot,
+                                    true,
+                                )
+                                .await
+                                .map_err(|error| {
+                                    SpComponentUpdateTerminalError::SetRotActiveSlotFailed {
```

Comment:
Tiny nit - should this be a new `SetRotBootloaderActiveSlotFailed` variant to get a more precise error message?

Reply:
I'll make that change